aws/config.go

```diff
@@ -239,6 +239,13 @@ type Config struct {
 	//    	Key: aws.String("/foo/bar/moo"),
 	//    })
 	EnableEndpointDiscovery *bool
+
+	// DisableEndpointHostPrefix will disable the SDK's behavior of prefixing
+	// request endpoint hosts with modeled information.
+	//
+	// Disabling this feature is useful when you want to use local endpoints
+	// for testing that do not support the modeled host prefix pattern.
+	DisableEndpointHostPrefix *bool
```

Comment:
Since the feature is enabled by default, a negative flag is needed to disable it. Like you mention the SDK uses the negative pattern to disable functionality that is enabled by default.